src/util/newconfig/ConfigValue.hpp

```diff
@@ -0,0 +1,189 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2024, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#pragma once
+
+#include "util/Assert.hpp"
+
+#include <cstddef>
+#include <cstdint>
+#include <optional>
+#include <string>
+#include <type_traits>
+#include <variant>
+
+namespace util::config {
+
+template <typename>
+constexpr bool always_false = false;
+
+/** @brief Custom clio config types */
+enum class ConfigType { Integer, String, Double, Boolean };
+
+/**
+ * @brief Get the corresponding clio config type
+ *
+ * @tparam Type The type to get the corresponding ConfigType for
+ * @return The corresponding ConfigType
+ */
+template <typename Type>
+constexpr ConfigType
+getType()
+{
+    if constexpr (std::is_same_v<Type, int>) {
+        return ConfigType::Integer;
+    } else if constexpr (std::is_same_v<Type, std::string>) {
+        return ConfigType::String;
+    } else if constexpr (std::is_same_v<Type, double>) {
+        return ConfigType::Double;
+    } else if constexpr (std::is_same_v<Type, bool>) {
+        return ConfigType::Boolean;
+    } else {
+        static_assert(always_false<Type>, "Wrong config type");
+    }
+}
+
+/**
+ * @brief Represents the config values for Json/Yaml config
+ *
+ * Used in ClioConfigDefinition to indicate the required type of value and
+ * whether it is mandatory to specify in the configuration
+ */
+class ConfigValue {
+    friend class ValueView;
+
+public:
+    using Type = std::variant<int, std::string, bool, double>;
+
+    constexpr ConfigValue() = default;
```

Comment:
Is default ConfigValue a required intType?

Reply:
Don't think the default is needed. 👍  Originally was trying to make everything constexpr and forgot to delete it

Reply:
Let's delete it then? 🙂

Reply:
Used in file to return empty configValue now: https://github.com/XRPLF/clio/pull/1544/files#diff-4e8d0eeb9ae15e14a017e6d321f3987f1c1e411fd58bb6eea9540f715328818eR178